custom/icds/tests/test_custom_rules.py

```diff
@@ -0,0 +1,128 @@
+import pytz
+from corehq.apps.data_interfaces.models import AutomaticUpdateRule, CustomActionDefinition
+from corehq.apps.data_interfaces.tests.test_auto_case_updates import BaseCaseRuleTest
+from corehq.apps.data_interfaces.tests.util import create_case, create_empty_rule
+from corehq.apps.hqcase.utils import update_case
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from corehq.form_processor.tests.utils import run_with_all_backends
+from corehq.util.timezones.conversions import ServerTime
+from datetime import datetime
+
+
+class AutoEscalationTest(BaseCaseRuleTest):
+    domain = 'icds-auto-escalation-test'
+
+    @property
+    def todays_date(self):
+        date = ServerTime(datetime.utcnow()).user_time(pytz.timezone('Asia/Kolkata')).done().date()
+        return date.strftime('%Y-%m-%d')
+
+    def _test_auto_escalation(self, from_level, to_level):
+        rule = create_empty_rule(self.domain, AutomaticUpdateRule.WORKFLOW_CASE_UPDATE)
+        rule.add_action(CustomActionDefinition, name='ICDS_ESCALATE_TECH_ISSUE')
+
+        with create_case(
+            self.domain,
+            'tech_issue',
+            case_name='New Issue',
+            update={
+                'ticket_level': from_level,
+                'touch_case_date': '2017-06-01',
+                'block_location_id': 'block_id',
+                'district_location_id': 'district_id',
+                'state_location_id': 'state_id',
+            },
+        ) as tech_issue:
+            properties = tech_issue.to_json()
+            self.assertEqual(properties.get('ticket_level'), from_level)
+            self.assertEqual(properties.get('touch_case_date'), '2017-06-01')
+            self.assertIsNone(properties.get('change_in_level'))
+
+            result = rule.run_actions_when_case_matches(tech_issue)
+            self.assertEqual(result.num_updates, 1)
+            self.assertEqual(result.num_creates, 1)
+
+            tech_issue = CaseAccessors(self.domain).get_case(tech_issue.case_id)
+            properties = tech_issue.to_json()
+            self.assertEqual(properties.get('ticket_level'), to_level)
+            self.assertEqual(properties.get('touch_case_date'), self.todays_date)
+            self.assertEqual(properties.get('change_in_level'), '1')
+
+            subcases = tech_issue.get_subcases(index_identifier='parent')
+            self.assertEqual(len(subcases), 1)
+            [tech_issue_delegate] = subcases
+
+            self.assertEqual(tech_issue_delegate.type, 'tech_issue_delegate')
+            self.assertEqual(tech_issue_delegate.name, tech_issue.name)
+            self.assertEqual(tech_issue_delegate.owner_id,
+                tech_issue.get_case_property('%s_location_id' % to_level))
+            self.assertEqual(tech_issue_delegate.get_case_property('change_in_level'), '1')
+
+    @run_with_all_backends
```

Comment:
I don't think it's necessary to run these with against couch backend since ICDS is SQL. Probably better to mark the test class with `@use_sql_backend`